fast64_internal/mk64/mk64_actors.py

```diff
@@ -0,0 +1,200 @@
+import math, bpy, mathutils
+import os
+from bpy.utils import register_class, unregister_class
+from ..panels import MK64_Panel
+
+
+
+enumModelIDs = [
+    ("Custom", "Custom", "Custom"),
+    ("ITEMBOX", "itembox", "itembox"),
+    ("FOLIAGE", "foliage", "foliage"),
+    ("FALLING_ROCK", "fallingRocks", "fallingRocks"),
+    ("PALM_TREES", "palmTrees", "palmTrees"),
+    ("PIRANHA_PLANT", "piranhaPlant", "piranhaPlant"),
+    ("Cow", "cow", "cow"),
+]
+
+class SearchModelIDEnumOperator(bpy.types.Operator):
+    bl_idname = "object.search_model_id_enum_operator"
+    bl_label = "Search Model IDs"
+    bl_property = "mk64_model_enum"
+    bl_options = {"REGISTER", "UNDO"}
+
+    mk64_model_enum: bpy.props.EnumProperty(items=enumModelIDs)
+
+    def execute(self, context):
+        context.object.mk64_model_enum = self.mk64_model_enum
+        bpy.context.region.tag_redraw()
+        self.report({"INFO"}, "Selected: " + self.mk64_model_enum)
+        return {"FINISHED"}
+
+    def invoke(self, context, event):
+        context.window_manager.invoke_search_popup(self)
+        return {"RUNNING_MODAL"}
+
+
+class MK64ObjectPanel(bpy.types.Panel):
+    bl_label = "Object Inspector"
+    bl_idname = "OBJECT_PT_MK64_Object_Inspector"
+    bl_space_type = "PROPERTIES"
+    bl_region_type = "WINDOW"
+    bl_context = "object"
+    bl_options = {"HIDE_HEADER"}
+
+    @classmethod
+    def poll(cls, context):
+        return context.scene.gameEditorMode == "MK64" and (
+            context.object is not None and context.object.type == "EMPTY"
+        )
+
+    def draw_inline_obj(self, box: bpy.types.UILayout, obj: bpy.types.Object):
+        obj_details: InlineGeolayoutObjConfig = inlineGeoLayoutObjects.get(obj.mk64_obj_type)
+
+        # display transformation warnings
+        warnings = set()
+        if obj_details.uses_scale and not obj_scale_is_unified(obj):
+            warnings.add("Object's scale must all be the same exact value (e.g. 2, 2, 2)")
+
+        if not obj_details.uses_scale and not all_values_equal_x(obj.scale, 1):
+            warnings.add("Object's scale values must all be set to 1")
+
+        loc = obj.matrix_local.decompose()[0]
+        if not obj_details.uses_location and not all_values_equal_x(loc, 0):
+            warnings.add("Object's relative location must be set to 0")
+
+        if not obj_details.uses_rotation and not all_values_equal_x(obj.rotation_euler, 0):
+            warnings.add("Object's rotations must be set to 0")
+
+        if len(warnings):
```

Comment:
this is nothing and can probably be ignored but personally I don't like doing ``if len(x)``, I prefer ``if len(x) > 0`` to make the intention clear